renku/service/jobs/contexts.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Context for jobs."""
+import contextlib
+import time
+
+from renku.service.jobs.queues import WorkerQueues
+
+
+@contextlib.contextmanager
+def enqueue_retry(queue, retry=3):
+    """Ensure job gets queued."""
+    count = 0
+    while count < retry:
+        try:
+            yield WorkerQueues.get(queue)
```

Comment:
The way I see it, those 2 things are achieving different things, setting this on Redis connection object will definitely add robustness to connection handling and I agree we should have it, but it doesn't address the problem which is being address with this context manager.

As far as I can see `TimeoutError` is not `OSError` nor `IOError`, so handling general network problems is not handled anywhere in both libraries and are left to us to handle. This context manager should cover those and busy loading case.

Thanks for suggestion though, I added it to Redis object and corresponding environment variable to manage the check interval.

Reply:
Oh I didn't mean use that instead, but in addition to what we're doing here, I should have been more clear.

A more general way to deal with this would be nice, maybe in the future we can extend `redis.Redis` with this and use rq's connection stack to pick up the connection via the `Connection` context manager.